app/models/concerns/inventory_computer.rb

```diff
@@ -75,13 +76,25 @@ def total_quantity
         historical_quantity.as_of_now
       end
 
+      def on_hand_packed_quantities_for(package)
+        res = historical_quantity.joins("INNER JOIN packages on packages.id = packages_inventories.source_id and packages_inventories.source_type = 'Package'")
+        res = res.select('packages_inventories.*')
+        res = res.where(packages_inventories: { package_id: package.id,
+                                                action: PACK_UNPACK })
+        res = res.where.not(packages: { on_hand_quantity: 0 })
+                 .group('packages.storage_type_id').raw.sum(:quantity)
+        res = storage_type.group_by(res, :name)
+        res = box_pallet_on_hand_qty_hash(res)
+        res
```

Comment:
no point in using `historical_quantity` if you do a `raw` and `sum` after. You can just use a normal query